gcapi/gcapi.py

```diff
@@ -539,6 +538,123 @@ def __call__(
         else:
             return response
 
+    def get_algorithm(self, algorithm: str) -> dict:
+        """Get the algorithm for the given algorithm name. """
+        algorithms = [
+            a
+            for a in list(
+                self.algorithms.list(params={"slug": algorithm})["results"]
+            )
+        ]
+
+        if len(algorithms) != 1:
+            raise ValueError(
+                f"{algorithm} is not found in available list of algorithms"
+            )
+
+        return algorithms[0]
+
+    def _upload_files(self, files):
+        raw_image_upload_session = self.raw_image_upload_sessions.create()
+
+        uploaded_files = {}
+        for file in files:
+            uploaded_chunks = self.chunked_uploads.upload_file(file)
+            uploaded_files.update(
+                {c["uuid"]: c["filename"] for c in uploaded_chunks}
+            )
+
+        for file_id, filename in uploaded_files.items():
+            self.raw_image_upload_session_files.create(
+                upload_session=raw_image_upload_session["api_url"],
+                staged_file_id=file_id,
+                filename=filename,
+            )
+
+        return raw_image_upload_session
+
+    def run_external_job(self, *, algorithm: str, inputs: Dict[str, Any]):
```

Comment:
`Any` is probably too broad, maybe `Union[List[Path],str]` (where the str is actually a URL to an image? or upload session? or a pk?), also should have a return type of algorithm job. Not worth fixing as types right now are a huge rabbit hole for little gain here.